### PR TITLE
Migrate to null safety

### PR DESCRIPTION
**Pull request by robbecker-wf**
Converts to null safety.

QA by using it in other packages. Tests should pass on Dart 2.18, 2.19. Tests won't pass on Dart 3 yet because we have tests that run on non-null safe packages.

### CODE REVIEW

**Comment by aviary2-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

**Comment by autobot-wk** [comment]
Skynet test results failed initially for this build but were approved by rob.becker
https://wf-skynet-hrd.appspot.com/apps/test/smithy/4349089/1
Approval message: no github actions need verifying

File: test/functional/documentation_test.dart [hunk 1]
```diff
@@ -73,14 +73,14 @@ String pubspecWithPackages(Set<String> packages) {
 
 class DartBlock {
   final int index;
-  final Set<String> packages;
+  final Set<String?> packages;
```
Comment (evanweible-wf) [inline test/functional/documentation_test.dart:76]: I think we can make this non-nullable if we fix up the parsing below

**Review by evanweible-wf — CHANGES_REQUESTED** [review]

**Review by evanweible-wf — APPROVED** [review]

**Comment by robbecker-wf** [comment]
QA+1 CI Passes and manually tested in a few other packages. Tried test, analyze, format, and server tasks.

**Comment by evanweible-wf** [comment]
semver +1
@Workiva/release-management-p

**Review by rmconsole-wf — APPROVED** [review]
+1 from RM

**Pull request merged**